### PR TITLE
Kylea/7.x

### PR DESCRIPTION
**Pull request by knjackson30974**

### CODE REVIEW

File: includes/tn_upload.form.inc [hunk 1]
```diff
@@ -71,11 +70,12 @@ function islandora_entities_tn_upload_form_submit(array $form, array &$form_stat
 
   $tn_file = file_load($form_state['values']['file']);
   if (isset($tn_file->uri)) {
-    islandora_temp_file_entry($tn_file->uri);
     $tn_path = drupal_realpath($tn_file->uri);
-    $image = image_load($tn_path);
```
Comment (jordandukart) [inline includes/tn_upload.form.inc:76]: Not sure this should be changing?

File: includes/callbacks.inc [hunk 2]
```diff
@@ -64,19 +64,19 @@ function islandora_entities_autocomplete($type) {
   $mappings = array(
     'scholar' => array(
       'cmodel' => 'islandora:personCModel',
-      'title' => 'MADS_title_mt',
+      'title' => 'islandora_entities_title_collection',
```
Comment (jordandukart) [inline includes/callbacks.inc]: These should reference the new variables such as `variable_get('islandora_entities_scholar_title_solr_field, 'MADS_title_mt')`.

File: includes/admin.form.inc [hunk 3]
```diff
@@ -92,5 +92,49 @@ function islandora_entities_settings($form, &$form_state) {
     '#required' => TRUE,
   );
 
+  $form['islandora_entities_title_collection'] = array(
```
Comment (jordandukart) [inline includes/admin.form.inc]: These aren't actually referencing collections but instead Solr fields used for autocomplete, variable names such as `islandora_entities_scholar_title_solr_field` probably makes sense.

File: includes/admin.form.inc [hunk 4]
```diff
@@ -92,5 +92,49 @@ function islandora_entities_settings($form, &$form_state) {
...
+    '#type' => 'textfield',
+    '#title' => t('Collection to ingest title entities in.'),
+    '#default_value' => variable_get(
+      'islandora_entities_title_collection',
+      'islandora:entity_collection'
+    ),
+    '#description' => t('Used by dependent modules provide links to the collection.'),
```
Comment (jordandukart) [inline includes/admin.form.inc]: The descriptions should be updated to note about being used for autocompletion.

Comment (knjackson30974) [inline includes/admin.form.inc]: Should I also change the titles on them?

File: includes/admin.form.inc [hunk 5]
```diff
@@ -92,5 +92,49 @@ function islandora_entities_settings($form, &$form_state) {
...
+      'islandora:entity_collection'
+    ),
+    '#description' => t('Used by dependent modules for autocompletion.'),
+    '#required' => TRUE,
+  );
+
+  $form['islandora_entities_last_name_solr_field'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Collection to ingest family name entities in.'),
+    '#default_value' => variable_get(
```
Comment (willtp87) [inline includes/admin.form.inc:131]: All variable gets on a given variable should have the same default.

**Comment by willtp87** [comment]
New variables need to be added to the uninstall hook https://github.com/Islandora/islandora_solution_pack_entities/blob/7.x/islandora_entities.install#L24

File: includes/admin.form.inc (hunk 4 above)
Comment (willtp87) [inline includes/admin.form.inc]: Yes.

File: includes/admin.form.inc [hunk 6]
```diff
@@ -92,5 +92,49 @@ function islandora_entities_settings($form, &$form_state) {
...
+  );
+
+  $form['islandora_entities_disambiguated_solr_field'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Collection to ingest disambiguated full name entities in.'),
+    '#default_value' => variable_get(
+      'islandora_entities_disambiguated_solr_field',
+      'islandora:entity_collection'
+    ),
+    '#description' => t('Used by dependent modules for autocompletion.'),
```
Comment (willtp87) [inline includes/admin.form.inc]: As per https://github.com/Islandora/islandora_solution_pack_entities/pull/74#discussion_r28295158 not sure the dependent module business is completely accurate; they are used in this module as well?

**Comment by willtp87** [comment]
I'm indifferent borderline bug/improvement.

**Comment by willtp87** [comment]
Sure, but it says bug because it wouldn't let me autocomplete improvement.

**Comment by willtp87** [comment]
A bug's fine, but is there a reason it didn't work or bug?

**Comment by willtp87** [comment]
Yeah Jira, wouldn't let me switch to an improvement.  I think 75 is Alan's gig yeah?

**Pull request merged**